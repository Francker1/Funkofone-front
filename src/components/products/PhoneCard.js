import React from 'react';
import PropTypes from 'prop-types';
import { Col, Row, Card } from 'react-bootstrap';
import { Link } from 'react-router-dom';
import BtnBubble from '../buttons/bubble/BtnBubble';

import './style/phone-card.css';

const PhoneCard = (props) => {
  return (
    <Col sm={4} className="mb-5">
      <Card className="shadow border-0">
        <Card.Img variant="top" src="https://picsum.photos/id/237/250/150" />
        <h6 className="card-maker">Apple</h6>
        <Card.Body>
          <h4>Phone Name</h4>
          <p>This is a wider card with supporting text below as a natural lead-in to additional.</p>
          <p>color: white</p>
          <h5>50,00 â‚¬</h5>
        </Card.Body>
        <Row>
          <Col>
            <Link to="/phones/edit/15">Editar</Link>
          </Col>
          <Col>
            <BtnBubble url="/phones/15" text="Details" />
          </Col>
        </Row>
        <Card.Footer>
          <small className="text-muted">ID phone: 0001</small>
        </Card.Footer>
      </Card>
    </Col>
  );
};

PhoneCard.propTypes = {};

export default PhoneCard;
