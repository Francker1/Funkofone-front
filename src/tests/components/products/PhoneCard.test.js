import React from 'react';
import { shallow } from 'enzyme';
import PhoneCard from '../../../components/products/PhoneCard';

describe('Tests in PhoneCard component', () => {
  const data = {
    model: 'ASD',
    name: 'iPhone 11',
    _id: '01',
    manufacturer: 'Apple',
    color: 'red',
    price: 10,
    image: 'http://example.img',
  };

  const wrapper = shallow(
    <PhoneCard
      name={data.name}
      model={data.model}
      _id={data._id}
      manufacturer={data.manufacturer}
      color={data.color}
      price={data.price}
      image={data.image}
    />,
  );

  test('should render the component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should render <h4> tag with name phone', () => {
    const h4 = wrapper.find('h4');

    expect(h4.text().trim()).toBe(data.name);
  });

  test('should have an image with src and alt props', () => {
    const img = wrapper.find('CardImg');

    expect(img.props().src).toBe(data.image);
    expect(img.props().alt).toBe(data.name);
  });

  test('tag <small> should have class text-muted', () => {
    const small = wrapper.find('small');
    const className = small.prop('className');

    expect(className.includes('text-muted')).not.toBe(false);
  });
});
