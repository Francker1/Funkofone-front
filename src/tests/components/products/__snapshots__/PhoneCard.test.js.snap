// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in PhoneCard component should render the component correctly 1`] = `
<Col
  className="mb-5"
  lg={4}
  sm={6}
>
  <Card
    body={false}
    className="shadow border-0"
  >
    <CardImg
      alt="iPhone 11"
      height="230"
      src="http://example.img"
      variant="top"
    />
    <h6
      className="card-maker"
    >
      Apple
    </h6>
    <CardBody>
      <h4>
        iPhone 11
      </h4>
      <p>
        color: 
        red
      </p>
      <h5>
        10
         â‚¬
      </h5>
    </CardBody>
    <div
      className="d-flex mb-3 card-buttons"
    >
      <Col
        sm={6}
      >
        <BtnLinkDefault
          className="btn w-100"
          outlined="true"
          text="Edit"
          url="/phones/edit/01"
        />
      </Col>
      <Col
        sm={6}
      >
        <BtnLinkDefault
          className="btn w-100"
          outlined=""
          text="Details"
          url="/phones/01"
        />
      </Col>
    </div>
    <CardFooter>
      <small
        className="text-muted"
      >
        Model: 
        ASD
      </small>
    </CardFooter>
  </Card>
</Col>
`;
